sprintf.c

```diff
@@ -937,7 +937,7 @@ rb_str_format(int argc, const VALUE *argv, VALUE fmt)
     rb_str_tmp_frozen_release(orig, fmt);
     /* XXX - We cannot validate the number of arguments if (digit)$ style used.
      */
-    if (posarg >= 0 && nextarg < argc) {
+    if (posarg >= 0 && nextarg < argc && !(argc == 2 && RB_TYPE_P(argv[1], T_HASH))) {
```

Comment:
Doesn't this make `sprintf "test", {}` no warning?

Reply:
You think it should be a warning? I don't think it should be.

Reply:
It's OK if you don't mean keyword arguments by "passed as keywords".

Reply:
Right, I meant passing only a single hash as argument.